### DISCUSSION TITLE
Transaction submission not working for Valora

### DISCUSSION THREAD

**Discussion by uchibeke** [celo-org/react-celo#109] [General]
When moving to payment approval, Valora gets as far as asking for a pin. After the pin is entered, nothing further happens.

More details here https://github.com/celo-org/use-contractkit/issues/89

Tested on iOS firefox and chrome, and OSX Chrome.

I have tried using
- use-contractkit 3.0.0-beta
- contractkit 1.5.2

and 
- use-contractkit 3.0.0-beta
- contractkit 1.5.0 (since use-contractkit 3.0.0-beta depends on contractkit 1.5.0 )

Some other folks have reported the same issue on Discord
https://discord.com/channels/600834479145353243/929644242790383636/943560810725273671

**Comment by dckesler** [comment]
We are currently looking into this. Is there by chance an error message you could share with us?

**Reply by uchibeke** [reply]
No error message is returned in my terminal or from the network tab

It gets stuck on this screen even after accepting the transaction on Valora. Also Valora returns no error (It prompts me to go back the the app like the tnx was successful but it does not appear to be submitted)

<img width="1792" alt="Screen Shot 2022-02-17 at 7 37 45 PM" src="https://user-images.githubusercontent.com/5677552/154595350-afda61c5-6fab-43a3-b215-bd5b4fc39fb2.png">

**Comment by Bridgerz** [comment]
Getting similar reports from some users trying to sign transactions using Valora connected to the same devices mobile browser.

Some were only able to get transactions signed by connecting with their device's native browser (iOS + Safari & chrome + android).

Also reports of having to make sure their app is updated (which is a no brainer) but may be useful.

**Comment by jeanregisser** [comment]
@uchibeke could you tell us what version of Valora you used? and what network was this on?

**Comment by uchibeke** [comment]
I am using 1.25.2 and the latest iOS.

I have removed and re-added my account and can see the balance and receive funds

I have also uninstalled an reinstalled valora

I have tried both wifi and mobile network

So not sure what the issue is

So basically, I cannot send, swap or confirm transactions on Valora

This was the issue all along and not use-contractkit

**Reply by jeanregisser** [reply]
Thanks, your issue is indeed not directly with `use-contractkit`.
I think you bumped into a sneaky issue where a transaction with `gasPrice` set to 0 is stuck in the TX pool. Blocking all other transactions you're trying to send.

Please try the steps detailed in [this answer](https://github.com/dsrvlabs/celo-extension-wallet/issues/10#issuecomment-1020352366) and let us know how it goes.

**Reply by jeanregisser** [reply]
FYI, the steps above worked for @uchibeke 
> Uchi — Yesterday at 1:15 PM
> Hi Jean | Valora this works now. I sent a transaction from celo web wallet and the other transactions submitted after that
>
> I then sent another tnx from my dApp with gasPrice and it worked as expected!
